### ISSUE TITLE
'-fconcepts' gcc flag in compile_commands.json not ignored in clang10

### ISSUE THREAD

**Issue by Arghnews** [clangd/clangd#331]
Using `clangd 9.0.1` and a `compile_commands.json` that states to compile a hello world file with gcc and `-std=c++17 -fconcepts` with neovim and coc.nvim works perfectly.

Using the same setup as above but with `clangd 10.0.0` results in clangd displaying an error:
`[clang drv_unknown_argument] [E] Unknown argument: '-fconcepts'`

My understanding is this is simply due to gcc having the `-fconcepts` flag and clang not. As stated,  `clangd 9.0.1` ignores this flag and doesn't display this error message which is, in my opinion, definitely the preferred behaviour.

clangd version 10.0.0
NVIM v0.4.3 with coc.nvim v0.0.74
Ubuntu 18.10

**Comment by kadircet** [comment]
it is just an unknown compiler flag and hopefully shouldn't do any harm apart from showing a diagnostic for it. Is it effecting any other feature?

In an ideal world it would be nice to somehow suppress these diags for gcc only flags(or drop them) but i don't think we've got a complete mapping today, maybe we should start creating one.
But not surfacing diags when we can't create a compiler invocation seems like a no go as these are crucial when things go bad.

**Comment by Arghnews** [comment]
I appreciate the super fast response.

I had not considered the implications of "ignoring" it as fully as you've stated. My gcc8 manual says:

       -fconcepts
           Enable support for the C++ Extensions for Concepts Technical Specification, ISO 19217 (2015), which allows code like

                   template <class T> concept bool Addable = requires (T t) { t + t; };
                   template <Addable T> T add (T a, T b) { return a + b; }

I apologise if I don't understand your last sentence correctly, are you saying if clang and therefore clangd can't compile code with a flag it's dangerous to silently ignore the flag in case it changes whether or not the code should compile, or code meaning etc.?

**If** I understand correctly what you're saying, then as you say, unfortunately this flag most definitely can fundamentally change what constitutes a valid c++ program or not (and said program's behaviour), and so should not be ignored, or especially not silently anyway?

**Comment by kadircet** [comment]
> are you saying if clang and therefore clangd can't compile code with a flag it's dangerous to silently ignore the flag in case it changes whether or not the code should compile, or code meaning etc.?

right, it makes clang/clangd perform in a strange mode and tracing it back to compile commands is quite hard in the absence of those diags.

> If I understand correctly what you're saying, then as you say, unfortunately this flag most definitely can fundamentally change what constitutes a valid c++ program or not (and said program's behaviour), and so should not be ignored, or especially not silently anyway?

you are totally right.
`-fconcepts` is special though, in clang they should be turned on by default with `-std=c++20`.
so i assumed you were just annoyed by the fact that there's a diagnostic that's not going away.

**Comment by Arghnews** [comment]
Yeh the diagnostic is just annoying, that's all. Also I had assumed it was a regression since I've never seen it before with clang9. I'll just ignore it.
Thanks for the responses.
